marketplace/core/types/channels/whatsapp_base/mixins.py

```diff
@@ -0,0 +1,149 @@
+import abc
+import calendar
+from typing import TYPE_CHECKING
+from datetime import datetime
+
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.exceptions import ValidationError
+
+if TYPE_CHECKING:
+    from rest_framework.request import Request
+    from .interfaces import ProfileHandlerInterface, BusinessProfileHandlerInterface
+
+from marketplace.accounts.permissions import ProjectViewPermission
+from .requests.facebook import FacebookConversationAPI
+from .exceptions import FacebookApiException, UnableProcessProfilePhoto
+from .serializers import WhatsAppBusinessContactSerializer, WhatsAppProfileSerializer
+
+
+class QueryParamsParser(object):
+
+    QUERY_PARAMS_START_KEY = "start"
+    QUERY_PARAMS_END_KEY = "end"
+
+    DATE_FORMAT = "%m-%d-%Y"
+
+    ERROR_MESSAGE = "Parameter `{}` cannot be found or is invalid"
+
+    def __init__(self, query_params: dict):
+        self._query_params = query_params
+        self.start = self._parse_to_unix(self._get_start())
+        self.end = self._parse_to_unix(self._get_end())
+
+    def _parse_to_unix(self, time: datetime) -> str:
+        return calendar.timegm(time.utctimetuple())
+
+    def _get_start(self) -> datetime:
+        return self._get_param_datetime(self.QUERY_PARAMS_START_KEY)
+
+    def _get_end(self) -> datetime:
+        end = self._get_param_datetime(self.QUERY_PARAMS_END_KEY)
+        return end.replace(hour=23, minute=59, second=59)
+
+    def _get_param_datetime(self, key: str) -> datetime:
+        param = self._query_params.get(key, None)
+        try:
+            return datetime.strptime(param, self.DATE_FORMAT)
+        except (ValueError, TypeError):
+            self._raise(key)
+
+    def _raise(self, field: str):
+        raise ValidationError(self.ERROR_MESSAGE.format(field))
+
+
+class WhatsAppConversationsMixin(object):
+    @action(detail=True, methods=["GET"], permission_classes=[ProjectViewPermission])
+    def conversations(self, request: "Request", **kwargs) -> Response:
```

Comment:
Looking at rest_framework viewsets, i think parameter request is a Request object, not a string